verified_boot_main.md

```diff
@@ -66,7 +66,145 @@
+configuration mechanism that ensures no modification can be done to attached
+medium, enhancing security and integrity. Mounting a filesystem as read-only
+attaches storage with read-only configuration to Linux directory structure.
+This configuration acts as a additional layer of protection served by the
+system. Linux does allow to mount read-writeable filesystems (eg. ext4) to be
+mounted as read-only[^9].
+
+#### Read-only filesystem permission
+
+Read-only is a file-system permission, allowing for reading and copying the
+data, but prohibits modification and addons. A file, directory or
+[entire disk](#read-only-physical-media) can be `read-only`[^10].
+
+A read-only file is any file with read-only attribute enabled. Such files can
+be read but cannot be modified nor removed. A file might be read-only on a file
+level or directory level. If I directory read-only permission is set, all the
+files in directory inherit that permission[^11]. It is worth noting, that each
+filesystem might handle read-only permissions differently, thus "read-only"
+should be thought of as a concept, rather than mechanism. For example, NTFS
+supports six basic groups of permissions types that include[^12]: read, write,
+list folder contents, read & execute, modify and full control.
+
+#### Immutable storage concept
+
+Immutable storage is a term related to "read-only" concept. It is a storage
+protocol that ensures stored data cannot be altered within a set of indefinite
+amount of time. The term comes from object-oriented programming, which defines
+immutable objects as object which state cannot be changed after it's
+created. Immutability can be implemented at various levels of storage stacks,
+based on both hardware and software solutions[^13].
+
+One of the immutability implementations is WORM (Write Once, Read Many)
+principle. On the other hand, a direct implementation of WORM are
+[read-only physical media](#read-only-physical-media). WORM principle ensures
+once data is written it cannot be altered or removed. Immutability is also
+often related to technologies like snapshotting or immutable filesystems[^14].
+
+#### Immutable linux OS
+
+Immutable Linux operating systems are aimed to introduce reliable, more secure
+approach to Linux. Immutable Linux OS cores are designed to be unchangeable
+and read only, meaning any changes made to the system are lost when system
+reboots. The advantages of immutability are:
+* increased security - even if system is compromised on runtime, permanent
+changes cannot be made by design,
+* easy maintenance - updates are made via atomic upgrades.
+
+These types of systems are updated via creating new OS instance,
+deploying it and switching over to the new one. This process is referred to as
+"image-based-update"[^15]. Updates are done alongside reboot, the architecture
+ensures that in case of failure in updating the system, one can easily revert
+system to previous state. Immutable distributions use containerization for
```

Comment:
Some immutable distributions use containerization but not all. Containerization (as in docker / lxc) is optional.